src/test/shell/integration/startup_options_test.sh

```diff
@@ -82,4 +82,23 @@ function test_autodetect_server_javabase() {
   bazel --noautodetect_server_javabase version &> $TEST_log || fail "Should pass"
 }
 
+# Below are the regression tests for Issue #7489
+function test_multiple_bazelrc_later_overwrites_earlier() {
+  tmpdir1=$(cd $(mktemp -d) &> /dev/null && pwd -P)
+  tmpdir2=$(cd $(mktemp -d) &> /dev/null && pwd -P)
+  echo "common --repository_cache=$tmpdir1" > 1.rc
+  echo "common --repository_cache=$tmpdir2" > 2.rc
+  bazel --bazelrc=1.rc --bazelrc=2.rc info &> $TEST_log || fail "Should pass"
```

Comment:
Having some trouble with flag naming and internal test compatibility, using `"--${PRODUCT_NAME}rc=1.rc"` should fix it.

Reply:
Adjusted. Thanks for the pointer!